src/hotspot/share/logging/logFileStreamOutput.cpp

```diff
@@ -86,19 +87,49 @@ class FileLocker : public StackObj {
   }
 };
 
+static bool write_error_is_shown = false;
+
+bool LogFileStreamOutput::flush() {
+  bool result = true;
+  if (fflush(_stream) != 0) {
+    if (!write_error_is_shown) {
+      jio_fprintf(defaultStream::error_stream(),
+                  "Could not flush log: %s (%s (%d))\n", name(), os::strerror(errno), errno);
+      jio_fprintf(_stream, "\nERROR: Could not flush log (%d)\n", errno);
+      write_error_is_shown = true;
+    }
+    result = false;
+  }
+  return result;
+}
+
+#define WRITE_LOG_WITH_RESULT_CHECK(op, total)                \
```

Comment:
You may leave out total and reference "written" directly in the macro since its always used with "written". Leave it up to you though, its a matter of taste.

Reply:
I prefer that WRITE_LOG_WITH_RESULT_CHECK like a function, so I stay this change.